### PR TITLE
IF-519: Setup minio and s3 database

### PR DESCRIPTION
**Pull request by jsylupp**

### CODE REVIEW

**Comment by mattmikulina** [comment]
Task linked: [IF-519 Object Storage](https://app.clickup.com/t/2177764/IF-519)

File: fusionfrontend/src/app/fleet/components/content/asset-wizard/asset-wizard.component.ts [hunk 1]
```diff
@@ -110,6 +114,20 @@ export class AssetWizardComponent implements OnInit {
...
 
+  ngOnDestroy(): void {
+    if (this.ref) {
+      this.deleteUploadedImage();
+      this.ref.close();
+    }
+  }
+
+  private deleteUploadedImage() {
+    if (this.assetForm.get('imageKey')?.value !== ImageService.DEFAULT_ASSET_IMAGE_KEY) {
```
Comment (florian-paul) [inline fusionfrontend/src/app/fleet/components/content/asset-wizard/asset-wizard.component.ts:125]: Restrict the deleteFunction for CreateMode

File: fusionbackend/src/main/java/io/fusion/fusionbackend/service/storage/BaseClient.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        return content64Based.length() * 3L / 4L;
+    }
+
+    protected boolean isFileSizeInvalidBase64(final String content64Based,
+                                            final ObjectStorageConfiguration configuration) {
+        return getFileSizeFrom64Based(content64Based) > configuration.maxFileSizeMb * 1024 * 1024;
+    }
+
+    protected boolean isFileSizeInvalid(final byte[] content, final ObjectStorageConfiguration configuration) {
+        return content.length > configuration.maxFileSizeMb * 1024 * 1024;
```
Comment (florian-paul) [inline fusionbackend/src/main/java/io/fusion/fusionbackend/service/storage/BaseClient.java]: create variable for numbers

File: fusionfrontend/src/app/fleet/components/content/asset-wizard/asset-wizard.component.ts (hunk 1 above)
Comment (florian-paul) [inline fusionfrontend/src/app/fleet/components/content/asset-wizard/asset-wizard.component.ts:127]: Restrict the deleteFunction for CreateMode

**Review by florian-paul — CHANGES_REQUESTED** [review]
Well done, just little changes :)

**Pull request merged**